homeworks/week4/Huy_Anh.java

```diff
@@ -0,0 +1,171 @@
+import java.util.Arrays;
+// Given a list of intervals [low, high) where low < high and len(intervals) >= 1, uniformly randomly generate a point from the union of these intervals.
+// For example, given intervals [2,5], [1,3], method generateRandomValue produces a random value in [1, 2) with 20%, [2,3) with 40%, and [3,5) with 40%.
+
+// Clarify: In my opinion, if I uniformly generate numbers from each interval and add them up, I will get the correct solution without sorting.
+// For example, there is 2 intervals [1,3] and [2,4]. If I generate 1000 numbers uniformly from the first interval, there should be 500 numbers in
+// the [2,3]. The same thing happens in the second interval and I will get 500 numbers in [1,2], 500 numbers in [2,3] (1st interval) plus 500 number
+// [2,3] (2nd interval) and the last 500 numbers in [3,4]. Thus, naturally, I will get the result where [2,3] has double the prob.
```

Comment:
Your insight is correct. However this clause `if I uniformly generate numbers from each interval and add them up` is difficult to understand. When you find yourself explaining something difficult, consider starting with examples first and then ending with a conclusion.